webapp/src/containers/SettingsPage/saga.tsx

```diff
@@ -116,7 +116,14 @@ export function* updateSettings(action) {
       if (data.reindexAfterSaving) {
         yield put(restartModal());
         yield call(shutDownBinary);
-        yield call(restartNode, { isReindexReq: true });
+        if (data.deletePeersAndBlocks) {
```

Comment:
you can keep the same call here, and just modify the object that's being passed.

Reply:
Done